File: js/nova/primitive/number/Short.js
var Short = function () {
	this.value = 0;
	
	
};

Short.prototype.Short = function (value) {
	this.this(value);
};

Short.prototype.destroy = function () {
};

Short.prototype.this = function (value) {
	this = value;
};

Short.prototype.numDigits = function (number) {
	return numDigits(number);
};

Short.prototype.toString = function (value) {
	return toString(value);
};

Short.prototype.toString = function () {
	return this.toString(this.value);
};

Short.prototype.compareTo = function (other) {
	return this.value - other;
};

Short.prototype.multiply = function (value) {
	return this * value;
};

Short.prototype.super = function () {
};

File: java/nova/primitive/number/Short.java
package nova.primitive.number;

import java.util.Optional;
import nova.exception.ExceptionData;
import nova.exception.Exception;
import nova.exception.DivideByZeroException;
import nova.io.Console;
import nova.primitive.number.Number;
import nova.primitive.number.Byte;
import nova.primitive.number.Short;
import nova.primitive.number.Int;
import nova.primitive.number.Long;
import nova.primitive.number.Float;
import nova.primitive.number.Double;
import nova.primitive.Null;
import nova.primitive.number.Char;
import nova.primitive.Bool;
import nova.datastruct.list.Array;
import nova.datastruct.list.IntArray;
import nova.datastruct.list.CharArray;
import nova.datastruct.list.DoubleArray;
import nova.datastruct.list.IntRange;
import nova.thread.Thread;
import nova.thread.async.Async;
import nova.gc.GC;
import nova.math.Math;
import nova.Object;
import nova.String;
import nova.System;
import nova.Class;
import nova.datastruct.Comparable;
import nova.primitive.number.Integer;

public class NovaShort
{
	
	public short value;
	
	
	
	
	public NovaShort(short value)
	{
		init(value);
	}
	
	public void init(short value)
	{
		this.value = value;
	}
	
	public static int numDigits(short number)
	{
		return NovaLong.numDigits(number);
	}
	
	public static NovaString toString(short value)
	{
		return NovaLong.toString(value);
	}
	
	public NovaString toString()
	{
		return toString(value);
	}
	
	public short compareTo(short other)
	{
		return value - other;
	}
	
	public short multiply(short value)
	{
		return this.value * value;
	}
	
}
